pkg/api/dashboard.go

```diff
@@ -528,9 +528,15 @@ func (hs *HTTPServer) GetHomeDashboard(c *models.ReqContext) response.Response {
 		slugQuery := models.GetDashboardRefByIdQuery{Id: preference.HomeDashboardID}
 		err := hs.DashboardService.GetDashboardUIDById(c.Req.Context(), &slugQuery)
 		if err == nil {
-			url := models.GetDashboardUrl(slugQuery.Result.Uid, slugQuery.Result.Slug)
-			dashRedirect := dtos.DashboardRedirect{RedirectUri: url}
-			return response.JSON(http.StatusOK, &dashRedirect)
+			dq := models.GetDashboardQuery{Uid: slugQuery.Result.Uid, OrgId: c.OrgID}
+			err := hs.DashboardService.GetDashboard(c.Req.Context(), &dq)
+			dash := dtos.DashboardFullWithMeta{}
+			dash.Meta.CanEdit = c.SignedInUser.HasRole(org.RoleEditor)
```

Comment:
this is the same as we do when the dash is set via config (see line ~560), but feels a little wrong since things like the FolderTitle won't be set correctly. is there a better way to get a dashboard with the full DashMeta populated already?

Reply:
i've added the logic to get the folder name, but it still feels wonky. we basically want exactly the same logic that is in `GetDashboard` on line 96, but i don't wanna have to repeat that logic 😕 feels like there intentionally isn't a way to call that method, but then why do we have so much shared logic in it?!